wolfSSL/wolfssljni
# JNI: wrap wolfSSL_SessionIsSetup() in WolfSSLSession

This PR wraps the native wolfSSL API: `wolfSSL_SessionIsSetup(WOLFSSL_SESSION* session);` in the static Java method `WolfSSLSession.sessionIsSetup(long session)`.

The native session pointer can be obtained from a calling Java application by calling `WolfSSLSession.getSession()`.  `WolfSSLSession.getSession()` behaves like `wolfSSL_get1_session()` and returns a duplicate/copy of the `WOLFSSL_SESSION`. Calling applications must take care to free the session pointer when done by calling `WolfSSLSession.freeSession(long session)`.

Native `wolfSSL_SessionIsSetup()` was added after wolfSSL 5.7.0 with https://github.com/wolfSSL/wolfssl/pull/7430. If using native wolfSSL `LIBWOLFSSL_VERSION_HEX` in `wolfssl/version.h` is not greater than `0x05007000`, then `WOLFSSL_PR7430_PATCH_APPLIED` must be defined when compiling wolfssljni native JNI sources.  `java.sh` can be edited to define `WOLFSSL_PR7430_PATCH_APPLIED` in `cflags` if needed.

Other changes in this PR:
- Refactor `WolfSSLSessionTest` to use Junit test method annotations
- Add test method for `WolfSSLSession.get/set/freeSession()`
- In `WolfSSLSession.getSession()` JNI wrapper, check if session has ticket for TLS 1.3 before calling `wolfSSL_peek()`
- Correctly close client/server `SSLSocket` in `WolfSSLSocketTest.protocolConnectionTestExtendedSocket()`, prevents test hang with `--enable-jni --enable-all` on some JDK versions
- Fix Infer thread safety violation warnings in `WolfSSLSession` and `WolfSSLSocket`
- Remove Java 8 testing from Temurin JDK in GitHub Actions. It looks like Java 8 was removed for Mac OSX runners, error `Error: Could not find satisfied version for SemVer '8'.`.

ZD 17557
